### DISCUSSION TITLE
Firewall rules location

### DISCUSSION THREAD

**Discussion by Nuttymoon** [TOSIT-IO/TDP#59] [New components] [answered]
Firewall rules role and definition for all TDP services/components are currently being implemented in [tdp-collection](https://github.com/TOSIT-IO/tdp-collection). See:
- https://github.com/TOSIT-IO/tdp-collection/pull/541
- https://github.com/TOSIT-IO/tdp-collection/pull/567

The question has been raised about where the role and rules definition should be placed. We basically have 4 choices :
1. :smile: The `firewall` role in `tdp-collection` and the rules of each component in its collection (e.g. `hdfs_dn` rules in `tdp-collection` and `kafka_broker` rules in `tdp-collection-extras`)
2. :tada: The `firewall` role in `tdp-collection-extras` and the rules of each component in its collection (e.g. `hdfs_dn` rules in `tdp-collection` and `kafka_broker` rules in `tdp-collection-extras`)
3. :eyes: The `firewall` role and **all the rules** in `tdp-collection-extras`
4. :rocket: The `firewall` role and **all the rules** in a dedicated collection `tdp-collection-firewall`

IMO options :smile: (1.) and :eyes: (3.) make more sense as we co-locate the firewall rules with their components.

You can vote with: :smile: (1.) :tada: (2.) :eyes: (3.) :rocket: (4.) below :point_down:

**Comment by rpignolet** [comment, marked as answer]
3. :eyes: The `firewall` role and **all the rules** in `tdp-collection-extras`
